packages/search/graphql/resolvers/_queries/search.js

```diff
@@ -329,9 +330,20 @@ const getFirst = (first, filter, user, recursive, unrestricted) => {
   // making totalCount always available
   // querying a single document by path is always allowed
   const path = getFilterValue(filter, 'path')
+
   const repoId = getFilterValue(filter, 'repoId')
   const oneRepoId = repoId && (!Array.isArray(repoId) || repoId.length === 1)
-  if (DOCUMENTS_RESTRICT_TO_ROLES && !recursive && !path && !oneRepoId && !unrestricted) {
+
+  const format = getFilterValue(filter, 'format')
+  const whitelisted = format && format.length && DOCUMENTS_UNRESTRICTED_REPO_IDS && DOCUMENTS_UNRESTRICTED_REPO_IDS
+    .split(',')
```

Comment:
`DOCUMENTS_UNRESTRICTED_REPO_IDS` is split on each `getFirst` call, which is a bit over the top. Maybe prep variables for their use case on initialisation?